dali/operators/reader/loader/video/frames_decoder.cc

```diff
@@ -159,6 +160,7 @@ bool FramesDecoder::ReadRegularFrame(uint8_t *data, bool copy_to_output) {
       make_string("Failed to send packet to decoder: ", detail::av_error_string(ret)));
```

Comment:
being a bit pedantic about it, I think that to be safe we should do:
```
ret = avcodec_send_packet(av_state_->codec_ctx_, av_state_->packet_);
if (ret < 0) {
    auto errormsg = make_string("Failed to send packet to decoder: ", detail::av_error_string(ret)));
    av_packet_unref(av_state_->packet_);
    DALI_FAIL(errormsg);
}
```
An alternative would be to use an auto deleter

Reply:
Not sure, what is the difference here. If `DALI_ENFORE` throws we will call `av_packet_unref` in the destructor.

Reply:
```
using AVPacketPtr = std::unique_ptr<AVPacket, decltype(&av_packet_unref)>;
av_state_->packet_ = pkt_ptr(raw_ptr, av_packet_unref);
```
^^ you could use something like this to avoid leaks in general with those av classes around here

Reply:
All of them are grouped in the `AvState` and this is used in `FramesDecoder` vis unique pointer. So whatever happens with the `FramesDecoder` this av state is cleaned with its destructor where we do all the proper unrefs and cleanup.

Reply:
> Not sure, what is the difference here. If `DALI_ENFORE` throws we will call `av_packet_unref` in the destructor. Unless the exception is not caught, in which case it doesn't matter, since we are not calling any cleanup.

Fair point